tests/types/npm-test.json

```diff
@@ -254,6 +254,78 @@
       "expected_output": "pkg:npm/core@2.0.1#googleapis/api/annotations",
       "expected_failure": false,
       "expected_failure_reason": null
+    },
+    {
+      "description": "Valid npm package without scope",
+      "test_group": "base",
+      "test_type": "validation",
+      "input": {
+        "type": "npm",
+        "namespace": null,
+        "name": "foobar",
+        "version": "12.3.1",
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_messages": null
+    },
+    {
+      "description": "Valid npm package with scope",
+      "test_group": "base",
+      "test_type": "validation",
+      "input": {
+        "type": "npm",
+        "namespace": "%40angular",
+        "name": "animation",
+        "version": "12.3.1",
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_messages": null
+    },
+    {
+      "description": "Valid npm package with qualifier vcs_url",
+      "test_group": "advanced",
+      "test_type": "validation",
+      "input": {
+        "type": "npm",
+        "namespace": null,
+        "name": "mypackage",
+        "version": "12.4.5",
+        "qualifiers": {
+          "vcs_url": "git://host.com/path/to/repo.git%404345abcd34343"
+        },
+        "subpath": null
+      },
+      "expected_messages": null
+    },
+    {
+      "description": "Invalid npm package with uppercase name",
+      "test_group": "base",
+      "test_type": "validation",
+      "input": {
+        "type": "npm",
+        "namespace": null,
+        "name": "MyPackage",
+        "version": "1.0.0",
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_messages": ["Name is not lowercased for purl type: 'npm'"]
```

Comment:
This is wrong. NPM does support mixed case names. PURL needs to stop adding new files saying that it doesn't.

Reply:
@matt-phylum You are correct that npm supported mixed case names in the past. But this is has long been not the case, like for the last 10+ years.

The current npm package.json spec is clear on this. Please see: https://docs.npmjs.com/cli/v11/configuring-npm/package-json

> New packages must not have uppercase letters in the name. 

See full details on that at:
- https://github.com/package-url/purl-spec/issues/136#issuecomment-3240215858

@TG1999 until this is sorted out, can you move this one validation test to a separate PR?